projects/scripts/project-xilinx.mk

```diff
@@ -115,12 +114,9 @@ $(PROJECT_NAME).sdk/system_top.xsa: $(M_DEPS)
 		$(PROJECT_NAME)_vivado.log, \
 		$(HL)$(PROJECT_NAME)$(NC) project)
 
-lib:
-	@for lib in $(LIB_DEPS); do \
-		if [ -n "${REQUIRED_VIVADO_VERSION}" ]; then \
-			$(MAKE) -C $(HDL_LIBRARY_PATH)$${lib} xilinx REQUIRED_VIVADO_VERSION=${REQUIRED_VIVADO_VERSION} || exit $$?; \
-		else \
-			$(MAKE) -C $(HDL_LIBRARY_PATH)$${lib} xilinx || exit $$?; \
-		fi; \
-	done
-
+$(HDL_LIBRARY_PATH)%/component.xml:
```

Comment:
This change causes the project to be unable to track library changes.
Library dependencies are managed in `library.mk`, so we should always run make on all library dependencies and let `library.mk` manage and skip them.

I will commit this change to your branch, if you do not like it, feel free to revert it.

(The other issue with "a library that failed will not be rebuilt" I'll leave to @liambeguin and @AndreiGrozav, as I have no preference for either solution).

Reply:
Isn't this related to what @AndreiGrozav was saying? about the fact that `flock` is creating a file even if the build fails?

Reply:
It's slightly different.
Andrei problem: a failed component.xml build have a newer timestamp than the dependencies (library.mk level).
My problem: if component.xml file exists, it never rebuilds the library even if the dependencies changes (xilinx-project.mk level).